src/components/JdsIcon/Icon.vue

```diff
@@ -0,0 +1,113 @@
+<template>
+  <i
+    :class="iconClasses"
+    v-on="$listeners"
```

Comment:
found a litte bit confusing where variable `$listeners` came from ?
i can't see that variable define on below, do you explain about that ?

Reply:
`$listeners` can be found [here](https://vuejs.org/v2/api/#vm-listeners).

`$listeners` can be used on root element to make a transparent wrapper for event listeners.
All event listeners define on its parent will automatically be assigned to the component,
without having to manually call $emit.

In parent/consumer,
```html
<template>
   <jds-icon @click="onClick" @focus="onFocus" @mouseover="onMouseOver" @keydown="onKeydown" />
</template>
``` 

In JdsIcon
```html
<template>
   <i class="iconClasses" v-on="$listeners">
       <!-- this root element will listen to click, focus, mouseover, keydown, as defined in its parent -->
   </i>
</template>
```

Reply:
without using v-on="$listeners",
we usually have to use `native` modifier on its parent/consumer.

In JdsIcon
```html
<template>
   <i class="iconClasses" >
       <!-- this root element is not using $listeners -->
   </i>
</template>
```

In parent/consumer,
```html
<template>
   <jds-icon
     @click.native="onClick"
     @focus.native="onFocus"
     @mouseover.native="onMouseOver"
     @keydown.native="onKeydown"
   />
</template>
```

Reply:
That was great, cool 😱 i just found out about it